readers.py

```diff
@@ -95,51 +94,52 @@ class TurtleReader():
-         'sound': 'sensor', 'volume': 'sensor', 'pitch': 'sensor',
-         'resistance': 'sensor', 'voltage': 'sensor', 'video': 'media',
-         'wait': 'media', 'camera': 'media', 'journal': 'media',
-         'audio': 'media', 'show': 'media', 'setscale': 'media',
-         'savepix': 'media', 'savesvg': 'media', 'mediawait': 'media',
-         'mediapause': 'media', 'mediastop': 'media', 'mediaplay': 'media',
-         'speak': 'media', 'sinewave': 'media', 'description': 'media',
-         'push':'extras', 'pop':'extras', 'printheap':'extras',
-         'clearheap':'extras', 'isheapempty2':'extras', 'chr':'extras',
-         'int':'extras', 'myfunction': 'python', 'userdefined': 'python',
-         'loadblock': 'python', 'loadpalette': 'python'}
+    TACAT = {'clean': 'forward', 'forward': 'forward', 'back': 'forward',
+             'left': 'forward', 'right': 'forward', 'arc': 'arc',
+             'xcor': 'coord', 'ycor': 'coord', 'heading': 'coord',
+             'setxy2': 'setxy', 'seth': 'setxy', 'penup': 'pen',
+             'pendown': 'pen', 'setpensize': 'pen', 'setcolor': 'pen',
+             'pensize': 'pen', 'color': 'pen', 'setshade': 'pen',
+             'setgray': 'pen', 'shade': 'pen', 'gray': 'pen',
+             'fillscreen': 'pen', 'startfill': 'fill', 'stopfill': 'fill',
+             'plus2': 'number', 'minus2': 'number', 'product2': 'number',
+             'division2': 'number', 'remainder2': 'number', 'sqrt': 'number',
+             'identity2': 'number', 'and2': 'boolean', 'or2': 'boolean',
+             'not': 'boolean', 'greater2': 'boolean', 'less2': 'boolean',
+             'equal2': 'boolean', 'random': 'random', 'repeat': 'repeat',
+             'forever': 'repeat', 'if': 'ifthen', 'ifelse': 'ifthen',
+             'while': 'ifthen', 'until': 'ifthen', 'hat': 'action',
+             'stack': 'action', 'storein': 'box', 'box': 'box',
+             'luminance': 'sensor', 'mousex': 'sensor', 'mousey': 'sensor',
+             'mousebutton2': 'sensor', 'keyboard': 'sensor',
+             'kbinput': 'sensor', 'readpixel': 'sensor', 'see': 'sensor',
+             'time': 'sensor', 'sound': 'sensor', 'volume': 'sensor',
+             'pitch': 'sensor', 'resistance': 'sensor', 'voltage': 'sensor',
+             'video': 'media', 'wait': 'media', 'camera': 'media',
+             'journal': 'media', 'audio': 'media', 'show': 'media',
+             'setscale': 'media', 'savepix': 'media', 'savesvg': 'media',
+             'mediawait': 'media', 'mediapause': 'media', 'mediastop': 'media',
+             'mediaplay': 'media', 'speak': 'media', 'sinewave': 'media',
+             'description': 'media', 'push': 'extras', 'pop': 'extras',
+             'printheap': 'extras', 'clearheap': 'extras',
+             'isheapempty2': 'extras', 'chr': 'extras', 'int': 'extras',
+             'myfunction': 'python', 'userdefined': 'python',
+             'loadblock': 'python', 'loadpalette': 'python'}
```

Comment:
Have these names been revalidated against the latest Turtle Art source?

Reply:
Yes, I have revalidated it from [here](https://github.com/sugarlabs/turtleart-activity/blob/master/pysamples/ta-stats.py)